### PR TITLE
Create justifications for all blocks

### PR DESCRIPTION
**Pull request by timorl**
# Description

Create justifications for all blocks. This not only fixes the linked bug (at least for validators), but also finally implements an ADR.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- I have made corresponding changes to the existing documentation
- I have created new documentation

### CODE REVIEW

**Comment by timorl** [comment]
I'm pretty sure the logic of status provider would now be even more natural if it iterated over the branch rather than properties, but this is urgent so I'm not touching that for now.

File: finality-aleph/src/data_io/status_provider.rs [hunk 1]
```diff
@@ -238,7 +248,7 @@ mod tests {
                 cached_cip,
                 aux_cip,
                 &proposal,
-                ProposalStatus::Finalize(proposal.top_block()),
+                ProposalStatus::Finalize(proposal.blocks_from_num(0).collect()),
```
Comment (DamianStraszak) [inline finality-aleph/src/data_io/status_provider.rs:251]: Would be great to have even a single test that checks the case when only some part of the branch is supposed to be finalized

**Review by DamianStraszak — CHANGES_REQUESTED** [review]
Looks good, although I requested a small test to be added, which would nicely accompany the changes.

**Review by piotrMocz — APPROVED** [review]
Looks great!

File: finality-aleph/src/data_io/status_provider.rs [hunk 2]
```diff
@@ -413,5 +413,15 @@ mod tests {
             &fresh_proposal,
             Finalize(fresh_proposal.blocks_from_num(0).collect()),
         );
+
+        // Long proposals should finalize the appropriate suffix.
+        let long_proposal = proposal_from_blocks(blocks[0..6].to_vec());
```
Comment (DamianStraszak) [inline finality-aleph/src/data_io/status_provider.rs:418]: Hmm, this seems to be the only line which mentions `long_proposal` so I'm a little suspicious :P.

Comment (timorl) [reply finality-aleph/src/data_io/status_provider.rs:418]: Whoopsie, should be better now.

**Review by DamianStraszak — APPROVED** [review]

**Pull request merged**